src/chunk.c

```diff
@@ -3404,7 +3404,7 @@ ts_chunk_drop_chunks(PG_FUNCTION_ARGS)
 	Dimension *time_dim;
 	Oid time_type;
 
-	PreventCommandIfReadOnly(DROP_CHUNKS_FUNCNAME "()");
+	PreventCommandIfReadOnly(psprintf("%s()", get_func_name(FC_FN_OID(fcinfo))));
```

Comment:
Maybe make a helper function for that and use it everywhere?

Reply:
I was actually considering a macro to avoid evaluating the function unless it really is read-only.

Reply:
Replaced it with a macro that also avoids executing the function unless it has to.